### PR TITLE
Add pulp_file plugin to pulpcore package

### LINKED ISSUE #4550: Add pulp_file APIs to the pulpcore package
Pulpcore 3.40.0 should include the File plugin. This will make it easier for developers to contribute tests that can only be written using a plugin with their pulpcore code changes.

The downside is that all users will be required to include the File plugin in their deployment. 

pulp_file docs need to be moved into pulpcore.

### PR DESCRIPTION
**Pull request by mdellweg**
fixes #4550

This is for 3.40.

### CODE REVIEW

File: functest_requirements.txt [hunk 1]
```diff
@@ -3,13 +3,10 @@ django
...
-pulpcore-client
-pulp-file-client
 pytest
 python-gnupg
 proxy.py~=2.4.3
 trustme~=1.1.0
 
-# Temporarily taken from pulp_file
-# Preferrably we can depend on pulp_file[test] instead.
+# pulp_file tests
```
Comment (dkliban) [inline functest_requirements.txt:11]: Did you mean to leave this comment?

**Comment by dkliban** [comment]
This looks good overall. The only thing that stands out to me is that we will now have functional tests in a 'from_plugin' module and a 'pulp_file' module. We should probably combine them in the future.

**Review by dkliban — APPROVED** [review]

File: functest_requirements.txt (hunk 1 above)
Comment (mdellweg) [reply functest_requirements.txt:11]: Yes, i did. We can remove it if wanted.

**Comment by mdellweg** [comment]
> This looks good overall. The only thing that stands out to me is that we will now have functional tests in a 'from_plugin' module and a 'pulp_file' module. We should probably combine them in the future.

I guess you mean using_plugin? I can even believe that we test the same things twice now. For sure there is a need to investigate and consolidate the test suite in the future.

File: .github/workflows/scripts/script.sh [hunk 2]
```diff
@@ -121,19 +121,11 @@ else
     if [[ "$GITHUB_WORKFLOW" == "Pulpcore Nightly CI/CD" ]] || [[ "${RELEASE_WORKFLOW:-false}" == "true" ]]; then
         cmd_user_prefix bash -c "pytest -v -r sx --color=yes --suppress-no-test-exit-code --pyargs pulpcore.tests.functional -m parallel -n 8 --nightly"
         cmd_user_prefix bash -c "pytest -v -r sx --color=yes --pyargs pulpcore.tests.functional -m 'not parallel' --nightly"
-        cmd_user_prefix bash -c "pytest -v -r sx --color=yes --suppress-no-test-exit-code --pyargs pulp_file.tests.functional -m parallel -n 8 --nightly"
-        cmd_user_prefix bash -c "pytest -v -r sx --color=yes --pyargs pulp_file.tests.functional -m 'not parallel' --nightly"
-        cmd_user_prefix bash -c "pytest -v -r sx --color=yes --suppress-no-test-exit-code --pyargs pulp_certguard.tests.functional -m parallel -n 8 --nightly"
```
Comment (ggainey) [inline .github/workflows/scripts/script.sh:126]: We aren't moving pulp_certguard (yet) - did you mean to remove these tests from pulpcore?

**Comment by decko** [comment]
Would make sense to squash `pulp_file` migrations here at this PR @mdellweg ?

File: ci_requirements.txt [hunk 3]
```diff
@@ -1,2 +0,0 @@
-pulp_file
-pulp-certguard
```
Comment (ggainey) [inline ci_requirements.txt:2]: We're removing pulp-certguard tests?

**Review by ggainey — COMMENTED** [review]
Overall, looks good, but I'm unclear on what's driving removing the certguard tests from pulpcore as part of this.

**Comment by decko** [comment]
> Would make sense to squash `pulp_file` migrations here at this PR @mdellweg ?

@ggainey brought two interesting historical documents for this:
1 - https://hackmd.io/@x9c4/rJbNGw-T5
2 - https://discourse.pulpproject.org/t/squashing-migrations/571

File: .github/workflows/scripts/script.sh (hunk 2 above)
Comment (mdellweg) [reply .github/workflows/scripts/script.sh:126]: Yes, at least temporarily. We intend to release pulpcore 3.40 soon and that will result in a version where we have no released certguard for. So we can bring those tests back when we have released certguard in a compatible version.

**Review by ggainey — APPROVED** [review]

**Pull request merged**